storage/bootstrap/bootstrapper/commitlog/source_index_test.go

```diff
@@ -0,0 +1,271 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package commitlog
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/m3db/m3db/persist/fs"
+	"github.com/m3db/m3db/persist/fs/commitlog"
+	"github.com/m3db/m3db/storage/bootstrap/result"
+	"github.com/m3db/m3db/storage/namespace"
+	"github.com/m3db/m3x/ident"
+	xtime "github.com/m3db/m3x/time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestBootstrapIndex(t *testing.T) {
+	var (
+		opts             = testOptions()
+		src              = newCommitLogSource(opts, fs.Inspection{}).(*commitLogSource)
+		dataBlockSize    = 2 * time.Hour
+		indexBlockSize   = 4 * time.Hour
+		namespaceOptions = namespace.NewOptions().
+					SetRetentionOptions(
+				namespace.NewOptions().
+					RetentionOptions().
+					SetBlockSize(dataBlockSize),
+			).
+			SetIndexOptions(
+				namespace.NewOptions().
+					IndexOptions().
+					SetBlockSize(indexBlockSize),
+			)
+	)
+	md, err := namespace.NewMetadata(testNamespaceID, namespaceOptions)
+	require.NoError(t, err)
+
+	now := time.Now()
+	start := now.Truncate(indexBlockSize)
+
+	fooTags := ident.Tags{ident.StringTag("city", "ny"), ident.StringTag("conference", "monitoroma")}
+	barTags := ident.Tags{ident.StringTag("city", "sf")}
+	bazTags := ident.Tags{ident.StringTag("city", "oakland")}
+	foo := commitlog.Series{Namespace: testNamespaceID, Shard: 0, ID: ident.StringID("foo"), Tags: fooTags}
```

Comment:
nit: could you use a var instead of `0` (and similarly in the the references below) and in the bootstrap request line (`targetRanges := result.ShardTimeRanges{0: ranges, 1: ranges, 2: ranges, 5: ranges}`). it'd be a lot clearer that way

Reply:
done (used shardZero + arithmetic to get the other shards)